test/e2e/dynamic_resource_e2e_test.go

```diff
@@ -0,0 +1,104 @@
+package e2e
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/registry"
+)
+
+func TestDynamicResourcesResolvePrometheusAPI(t *testing.T) {
+	defer cleaner.NotifyTestComplete(t, true)
+
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeConfigPath)
+	require.NoError(t, err)
+
+	c := newKubeClient(t)
+	crc := newCRClient(t)
+	dynamicClient := newDynamicClient(t, config)
+
+	ns, err := c.KubernetesInterface().CoreV1().Namespaces().Create(&corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: genName("ns-"),
+		},
+	})
+	require.NoError(t, err)
+
+	deleteOpts := &metav1.DeleteOptions{}
+	defer func() {
+		require.NoError(t, c.KubernetesInterface().CoreV1().Namespaces().Delete(ns.GetName(), deleteOpts))
+	}()
+
+	catsrc := &v1alpha1.CatalogSource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      genName("dynamic-catalog-"),
+			Namespace: ns.GetName(),
+		},
+		Spec: v1alpha1.CatalogSourceSpec{
+			Image:      "quay.io/olmtest/catsrc_dynamic_resources:e2e-test",
```

Comment:
Should we lock this down to a known digest to avoid breaking tests with changes?